pinot-core/src/main/java/org/apache/pinot/core/operator/filter/predicate/PredicateEvaluator.java

```diff
@@ -28,6 +28,16 @@
    * APIs for both dictionary based and raw value based predicate evaluator
    */
 
+  /**
+   * Set the predicate
+   */
+  void setPredicate(Predicate predicate);
```

Comment:
This should not be an interface method

Reply:
I see what you mean, but It is used in line 107 of `PredicateEvaluator.java` to pass predicate into `PredicateEvaluator`. Let me know if you have a better way to doing this.

Reply:
#7840

Reply:
Please merge the PR and I will pick it up here.